File: dongwook/week2/소수만들기/소수만들기.java
package Study_codingtest.dongwook.week2.소수만들기;

public class 소수만들기 {

    public int solution(int[] nums) {
        int answer = 0;

        // 3개의 수를 더한 값이 소수인지 확인하기 위한 루프
        for (int i = 0; i < nums.length - 2; i++) {
            for (int j = i + 1; j < nums.length - 1; j++) {
                for (int k = j + 1; k < nums.length; k++) {
                    int sum = nums[i] + nums[j] + nums[k]; // 세 수의 합
                    // 합이 소수인 경우 answer 값 증가시켜줌
                    if (isPrime(sum)) {
                        answer++;
                    }
                }
            }
        }
        return answer;
    }

    // 소수 판별 함수
    private boolean isPrime(int number) {
        if (number <= 1) {
            return false;
        }
        for (int i = 2; i * i <= number; i++) { //2부터 number의 제곱근까지 나눠서 소수를 판별하는 과정이다
            if (number % i == 0) {
                return false; // 나눠 떨어지면 false
            }
        }
        return true; // 소수인 경우
    }
}

File: Subin/week2/소수 만들기.py
from itertools import combinations

def solution(nums):
    answer =0
    combi = list(combinations(nums, 3)) #3개씩 모든 조합 저장
    
    check = [] #3개 합 리스트
    for x in range(len(combi)):
        check.append(sum(combi[x]))
    
    no=[] #소수가 아닌 리스트
    all = len(check)  # 전체 조합의 개수
   #소수 아닌거 판별
    for i in range(len(check)):
        if check[i]==1:
                no.append(1)
        else:
            for j in range(2,check[i]): 
                if check[i]%j==0:
                    no.append(check[i])
                    break
    return all - len(no)
                
    
    
